server/lib/matchmaking/matchmaking-times-api.ts

```diff
@@ -0,0 +1,147 @@
+import { RouterContext } from '@koa/router'
+import httpErrors from 'http-errors'
+import Joi from 'joi'
+import { container } from 'tsyringe'
+import { ALL_MATCHMAKING_TYPES, MatchmakingType } from '../../../common/matchmaking'
+import {
+  AddMatchmakingTimeRequest,
+  GetFutureMatchmakingTimesResponse,
+  GetPastMatchmakingTimesResponse,
+  MATCHMAKING_TIMES_LIMIT,
+  MatchmakingTimeJson,
+  toMatchmakingTimeJson,
+} from '../../../common/matchmaking/matchmaking-times'
+import { httpApi, httpBeforeAll } from '../http/http-api'
+import { httpDelete, httpGet, httpPost } from '../http/route-decorators'
+import {
+  addMatchmakingTime,
+  getCurrentMatchmakingTime,
+  getFutureMatchmakingTimes,
+  getMatchmakingTimeById,
+  getPastMatchmakingTimes,
+  removeMatchmakingTime,
+} from '../models/matchmaking-times'
+import { checkAllPermissions } from '../permissions/check-permissions'
+import ensureLoggedIn from '../session/ensure-logged-in'
+import { validateRequest } from '../validation/joi-validator'
+import MatchmakingStatusService from './matchmaking-status'
+
+function getValidatedMatchmakingType(ctx: RouterContext) {
+  const {
+    params: { matchmakingType },
+  } = validateRequest(ctx, {
+    params: Joi.object<{ matchmakingType: MatchmakingType }>({
+      matchmakingType: Joi.valid(...ALL_MATCHMAKING_TYPES).required(),
+    }),
+  })
+
+  return matchmakingType
+}
+
+@httpApi('/matchmaking-times')
+@httpBeforeAll(ensureLoggedIn, checkAllPermissions('manageMatchmakingTimes'))
+export class MatchmakingTimesApi {
+  @httpGet('/:matchmakingType/current')
+  async getCurrentTime(ctx: RouterContext): Promise<MatchmakingTimeJson | undefined> {
+    const matchmakingType = getValidatedMatchmakingType(ctx)
+
+    const current = await getCurrentMatchmakingTime(matchmakingType)
+
+    return current ? toMatchmakingTimeJson(current) : undefined
+  }
+
+  @httpGet('/:matchmakingType/future')
+  async getFutureTimes(ctx: RouterContext): Promise<GetFutureMatchmakingTimesResponse> {
+    const matchmakingType = getValidatedMatchmakingType(ctx)
+    const {
+      query: { offset },
+    } = validateRequest(ctx, {
+      query: Joi.object<{ offset: number }>({
```

Comment:
This typing isn't correct for what you've written here, you never made the parameter required. If it is required, mark it as such. If it isn't, mark it optional in this typing